src/System/reachability.rs

```diff
@@ -0,0 +1,199 @@
+
+    // hashmap linking every location to all its current zones
+    let mut visited_states: HashMap<LocationID, Vec<OwnedFederation>> = HashMap::new();
+    //let mut made_transitions: Vec<SubPath> = Vec::new();
+
+    // List of states that are to be visited
+    let mut frontier_states: Vec<Rc<SubPath>> = Vec::new();
+
+    let mut actions: Vec<String> = system.get_actions().into_iter().collect();
+    actions.sort();
+
+    visited_states.insert(
+        start_clone.get_location().id.clone(),
+        vec![start_clone.zone_ref().clone()],
+    );
+
+    frontier_states.push(Rc::new(SubPath {
+        previous_sub_path: None,
+        destination_state: start_clone,
+        transition: None,
+    }));
+
+    while let Some(sub_path) = frontier_states.pop() {
+        if reached_end_state(&sub_path.destination_state, end_state) {
+            return make_path(sub_path);
+        }
+
+        for action in &actions {
+            for transition in
+                &system.next_transitions(&sub_path.destination_state.decorated_locations, action)
+            {
+                take_transition(
+                    &sub_path,
+                    transition,
+                    &mut frontier_states,
+                    &mut visited_states,
+                    system,
+                );
+            }
+        }
+    }
+    // If nothing has been found, it is not reachable
+    Path {
+        path: None,
+        was_reachable: false,
+    }
+}
+
+fn reached_end_state(cur_state: &State, end_state: &State) -> bool {
+    cur_state
+        .get_location()
+        .id
+        .compare_partial_locations(&end_state.get_location().id)
+        && cur_state.zone_ref().has_intersection(end_state.zone_ref())
+}
+
+fn take_transition(
+    sub_path: &Rc<SubPath>,
+    transition: &Transition,
+    frontier_states: &mut Vec<Rc<SubPath>>,
+    visited_states: &mut HashMap<LocationID, Vec<OwnedFederation>>,
+    system: &dyn TransitionSystem,
+) {
+    let mut new_state = sub_path.destination_state.clone();
+    if transition.use_transition(&mut new_state) {
+        new_state.extrapolate_max_bounds(system); // Do we need to do this? consistency check does this
+        let new_location_id = &new_state.get_location().id;
+        let existing_zones = visited_states.entry(new_location_id.clone()).or_default();
+        if !zone_subset_of_existing_zones(new_state.zone_ref(), existing_zones) {
+            remove_existing_subsets_of_zone(new_state.zone_ref(), existing_zones);
+            visited_states
+                .get_mut(new_location_id)
+                .unwrap()
+                .push(new_state.zone_ref().clone());
+            frontier_states.push(Rc::new(SubPath {
+                previous_sub_path: Some(Rc::clone(sub_path)),
+                destination_state: new_state,
+                transition: Some(transition.clone()),
+            }));
+        }
+    }
+}
+
+/// Checks if this zone is redundant by being a subset of any other zone
```

Comment:
Sometimes you use `///`

Reply:
Code dokumentation, så det er muligt at se beskrivelsen af funktionen når man holder musen over den.
Vi bruger // når vi beskriver noget inde i en funktion til at gøre det nemmere at læse.